packages/amplify-cli-core/src/cfnUtilities.ts

```diff
@@ -112,13 +112,31 @@ const CF_SCHEMA = new yaml.Schema([
   new yaml.Type('!GetAtt', {
     kind: 'scalar',
     construct: function (data) {
-      return { 'Fn::GetAtt': Array.isArray(data) ? data : data.split('.') };
+      if (Array.isArray(data)) {
+        return {
+          'Fn::GetAtt': data,
+        };
+      }
+      // data is a string
+      const firstPeriodIdx = data.indexOf('.');
```

Comment:
If there is no `.` in the string, this will generate the wrong output. I'm not sure if it actually matters though since the `GetAtt` won't work properly.

Reply:
yeah I don't think we need to cover that case here as it's already invalid